### PR TITLE
DOC: Clarify virtualenv setup and dependency installation

### LINKED ISSUE #25166: DOC: Clarify `virtualenv` setup and dependency installation in the development environment guide
### Issue with current documentation:

Currently the [Using virtual environments](https://numpy.org/devdocs/dev/development_environment.html#using-virtual-environments) section of the contributing guide's "Setting up and using your development environment" includes helpful details for `conda` users about how to install development dependencies using the `environment.yml` file. However, the guide does not include corresponding details for `virtualenv` users and does not specify whether it's preferred to use `pyproject.toml` for this purpose or the suite of `requirements.txt` files. While specific subsections of the tutorial do include this step (for instance, the [Testing builds](https://numpy.org/devdocs/dev/development_environment.html#testing-builds) section instructs the user to first install test dependencies by running `python -m pip install -r test_requirements.txt`), there is no general instruction for `virtualenv` users for installing all project dependencies at once.

For more experienced developers, this process may be more obvious, but for beginners such as myself with less experience managing virtual environments, the added steps could be beneficial in onboarding new contributors more quickly.

### Idea or request for content:

I would like to propose fleshing out dependencies-installation instructions for `virtualenv` users (happy to volunteer) using either the `pyproject.toml` file or the `requirements.txt` files, as recommended by project maintainers.

For example, if maintainers recommend using the `requirements.txt` files for installing dependencies, resolving this issue could be as simple as expanding the `virtualenv` code snippet in the existing "Using virtual environments" section as follows:
```
$ virtualenv numpy-dev
$ python -m pip install -r build_requirements.txt
$ python -m pip install -r doc_requirements.txt
$ python -m pip install -r linter_requirements.txt
$ python -m pip install -r release_requirements.txt
$ python -m pip install -r test_requirements.txt 
```

### PR DESCRIPTION
**Pull request by zrottman**
Closes #25166.

Updates [Setting up and using your development environment](https://numpy.org/devdocs/dev/development_environment.html#setting-up-and-using-your-development-environment) with dependencies-installation instructions for `virtualenv` users. Now environment set up instructions for `virtualenv` match those of `conda`, which do specify steps for installing project dependencies.

Grateful for any feedback as this is my first contribution to NumPy.

### CODE REVIEW

**Review by timhoffm — COMMENTED** [review]
I suggest to use [sphinx-design tabs](https://sphinx-design.readthedocs.io/en/latest/tabs.html). We've done that at [matplotlib](https://matplotlib.org/stable/devel/development_setup.html#create-a-dedicated-environment), and it helps making clear that these are two mutually exclusive variants.

**Comment by rgommers** [comment]
> I suggest to use [sphinx-design tabs](https://sphinx-design.readthedocs.io/en/latest/tabs.html). We've done that at [matplotlib](https://matplotlib.org/stable/devel/development_setup.html#create-a-dedicated-environment), and it helps making clear that these are two mutually exclusive variants.

Agreed that sphinx-design is nice, but I'd much prefer to not do that here, since all these docs are getting a major overhaul very soon, because of the build system change. We'll copy the SciPy build docs as a base, which already use tabs.

File: doc/source/dev/development_environment.rst [hunk 1]
```diff
@@ -58,9 +58,16 @@ once)::
...
+virtualenv (named ``numpy-dev`` here), activate it, and install all project 
+dependencies with::
 
     $ virtualenv numpy-dev
+    $ source numpy-dev/bin/activate # activate virtual environment
+    $ python -m pip install -r build_requirements.txt
+    $ python -m pip install -r doc_requirements.txt
+    $ python -m pip install -r linter_requirements.txt
+    $ python -m pip install -r release_requirements.txt
+    $ python -m pip install -r test_requirements.txt
```
[the remark this answers is not in the document]
Comment (rgommers) [reply doc/source/dev/development_environment.rst:70]: this seems like a reasonable change to me.

File: doc/source/dev/development_environment.rst [hunk 2]
```diff
@@ -58,9 +58,12 @@ once)::
 
 If you installed Python some other way than conda, first install
 `virtualenv`_ (optionally use `virtualenvwrapper`_), then create your
-virtualenv (named ``numpy-dev`` here) with::
+virtualenv (named ``numpy-dev`` here), activate it, and install all project 
+dependencies with::
 
     $ virtualenv numpy-dev
+    $ source numpy-dev/bin/activate # activate virtual environment
```
Comment (rgommers) [inline doc/source/dev/development_environment.rst:65]: I'll note that there is no single cross-platform and cross-shell command to activate a virtualenv, this won't work on Windows. But I much prefer to not turn this into a longer venv tutorials, so let's leave it like this for now. I'll do a larger rewrite of all the dev docs the coming week anyway, if things go as planned.

**Review by rgommers — APPROVED** [review]
LGTM, thanks @zrottman!

**Pull request merged**